src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketWebhookRequest.java

```diff
@@ -0,0 +1,86 @@
+package com.atlassian.bitbucket.jenkins.internal.model;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+import static org.codehaus.groovy.runtime.InvokerHelper.asList;
+
+public class BitbucketWebhookRequest {
+
+    private final String name;
+    private final Set<String> events;
+    private final String url;
+    private final boolean isActive;
+
+    protected BitbucketWebhookRequest(BitbucketWebhookRequestBuilder builder) {
+        this.name = builder.name;
+        this.events = builder.events;
+        this.url = builder.url;
+        this.isActive = builder.isActive;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Set<String> getEvents() {
+        return events;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public boolean isActive() {
+        return isActive;
+    }
+
+    /**
+     * A builder to provide fluent way of building webhook register request.
+     */
+    public static class BitbucketWebhookRequestBuilder {
```

Comment:
I don't like this builder, my main objections are
1. `.build` is never invoked, it is only called from a test
2. As indicated below, there are required fields I must know about by reading code.
3. It makes no checks (in the non-test code path) against nulls, so it will happily pass `null` for name, and the `BitbucketWebhookRequest` will be created with a `null` name.

I think this needs to be changed to be safe.

Reply:
I have kept the builder as I am using it in next bit of work. I have introduced a constructor.